### PR TITLE
Members must open the network first 

### PR DESCRIPTION
**Pull request by jumaffre**
As part of https://github.com/microsoft/CCF/issues/293, this PR introduces member votes to open the network to users. This is a mandatory steps for any CCF network. 

When a CCF network is created, the SERVICE table is initialised with a new network in state OPENING. Then, only when members have successfully voted for the network to be opened, the SERVICE is marked as OPEN and a global hook guarantees the atomic transition to open the network to users.

Note that I slightly changed the schema of the SERVICE table so that only the current network is saved in the KV (at index 0, so no longer need to store the active service key in the VALUES table). 

Still a few things to do to complete this feature (to come in later PRs): https://github.com/microsoft/CCF/issues/293#issuecomment-530905069

### CODE REVIEW

**Comment by cimetrics** [comment]
images

File: tests/infra/ccf.py [hunk 1]
```diff
@@ -353,10 +370,16 @@ def vote(self, member_id, remote_node, proposal_id, accept, force_unsigned=False
             "vote",
             f"--proposal-id={proposal_id}",
             "--accept" if accept else "--reject",
-            "--sign" if not force_unsigned else "--force-unsigned",
```
[the remark this answers is not in the document]
Comment (achamayou) [reply tests/infra/ccf.py]: Yes it's default, because we reject them otherwise.

**Comment by cimetrics** [comment]
images

**Comment by olgavrou** [comment]
Can we add some tests for the cases where the network will not open due to invalid state? There seem to be quite a few.

**Comment by cimetrics** [comment]
images

**Comment by jumaffre** [comment]
> Can we add some tests for the cases where the network will not open due to invalid state? There seem to be quite a few cases where it will fail to open.

@olgavrou As discussed, I've added a new test in `member_client_test` to check that it is not possible to open a CCF network twice.

**Comment by cimetrics** [comment]
images

**Pull request merged**